helpers/test-data-generator.js

```diff
@@ -162,19 +162,14 @@ export async function setupFundedTask({
 
   const taskId = await makeTask({ colonyNetwork, colony, dueDate, domainId, skillId, manager });
   const task = await colony.getTask(taskId);
-  const { fundingPotId } = task;
   const managerPayoutBN = new BN(managerPayout);
   const evaluatorPayoutBN = new BN(evaluatorPayout);
   const workerPayoutBN = new BN(workerPayout);
   const totalPayouts = managerPayoutBN.add(workerPayoutBN).add(evaluatorPayoutBN);
 
+  const childSkillIndex = await getChildSkillIndex(colonyNetwork, colony, 1, task.domainId);
   await colony.setFundingRole(1, 0, manager, 1, true);
-  let toSkillIndex = 0;
-  if (task.domainId !== 1) {
-    // Then we need to figure out the correct skill indices.
-    toSkillIndex = await getChildSkillIndex(colonyNetwork, colony, 1, task.domainId);
-  }
```

Comment:
Removing this is what forced you into passing `colonyNetwork` in `makeTask` everywhere. This is because most tests use the root domain which has skill index of `0`

Reply:
Mm makes sense, but I'd advocate for keeping it the way it is in the PR for the following reasons:

1. It makes the `makeTask` interface consistent with the other task generation functions, all of which take `colonyNetwork` as the first argument.
2. It avoids the duplication of the domain-checking logic, which occurs already in `getChildSkillIndex`. Further, you could remove this check and _still_ pass an undefined `colonyNetwork` in cases where `domainId == 1` since `colonyNetwork` will never be used.
3. Further, I think that the logic which checks if the domains are the same should be hidden under the `getChildSkillIndex` abstraction layer; having that logic in `makeTask` seems like an unnecessary violation of abstraction boundaries.

That said, after merging #567 I'm pretty sure the only thing from this PR that's not elective are the updates to the `addDomain` invocations, so it's whatever you want really.